internal/collector/magnum/containerinfra.go

```diff
@@ -0,0 +1,57 @@
+package magnum
+
+import (
+	"database/sql"
+	"log/slog"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/vexxhost/openstack_database_exporter/internal/collector"
+)
+
+var (
+	magnumUpDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(collector.Namespace, Subsystem, "up"),
+		"up",
+		nil,
+		nil,
+	)
+)
+
+type ContainerInfraCollector struct {
+	db                *sql.DB
+	logger            *slog.Logger
+	clustersCollector *ClustersCollector
+	mastersCollector  *MastersCollector
+	nodesCollector    *NodesCollector
+}
+
+func NewContainerInfraCollector(db *sql.DB, logger *slog.Logger) *ContainerInfraCollector {
+	return &ContainerInfraCollector{
+		db:                db,
+		logger:            logger,
+		clustersCollector: NewClustersCollector(db, logger),
+		mastersCollector:  NewMastersCollector(db, logger),
+		nodesCollector:    NewNodesCollector(db, logger),
+	}
+}
+
+func (c *ContainerInfraCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- magnumUpDesc
+	c.clustersCollector.Describe(ch)
+	c.mastersCollector.Describe(ch)
+	c.nodesCollector.Describe(ch)
+}
+
+func (c *ContainerInfraCollector) Collect(ch chan<- prometheus.Metric) {
+	// Set up metric = 1 (service is up)
+	ch <- prometheus.MustNewConstMetric(
+		magnumUpDesc,
+		prometheus.GaugeValue,
+		1,
+	)
+
+	// Collect metrics from all sub-collectors
+	c.clustersCollector.Collect(ch)
+	c.mastersCollector.Collect(ch)
+	c.nodesCollector.Collect(ch)
+}
```

Comment:
The `_up` metric is always set to 1 here, but it should reflect whether the sub-collectors succeeded. Looking at `cinder/volumes.go`, the pattern is:
- Emit `up=0` immediately when a query fails and return early
- Emit `up=1` only at the end after all metrics are successfully collected

Consider having sub-collectors return an error, or track success state, so the umbrella collector can emit `up=0` when any sub-collector fails.